napari/plugins/_npe2.py

```diff
@@ -391,16 +392,22 @@ def _npe2_manifest_to_actions(
     return actions, submenus
 
 
+class _GroupOrder(TypedDict):
+    group: Optional[str]
+    order: Optional[float]
+    when: Optional[str]
+
+
 def _when_group_order(
     menu_item: contributions.MenuItem,
-) -> dict[str, Union[str, float, None]]:
+) -> _GroupOrder:
     """Extract when/group/order from an npe2 Submenu or MenuCommand."""
     group, _, _order = (menu_item.group or '').partition("@")
     try:
         order: Optional[float] = float(_order)
     except ValueError:
         order = None
-    return {'when': menu_item.when, 'group': group or None, 'order': order}
+    return _GroupOrder(when=menu_item.when, group=group or None, order=order)
```

Comment:
Thanks for the changes @Czaki with `_ExtraDict` and `_SummaryDict`; could we remove this also? I understand the utility but imho it is not worth the reduced readability. Code should first and foremost be easy to read. I've read the opinion that Java code is so hard to read because of their advanced tooling, and I'm starting to see Python go in that direction, and it's not a great feeling.

Reply:
👆 reminder

Reply:
Done